### PR TITLE
Add ConfigurationVersions

### PR DESCRIPTION
**Pull request by mortengjesing**
# Description

Added ConfigurationVersions methods.

Changed the client to handle Kabab case without adding `JsonPropertyName` attribute to every property

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [] Extended the documentation

### CODE REVIEW

File: src/Tfe.NetClient/Features/JsonKebabCaseNamingPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+using System.Text;
+using System.Text.Json;
+
+namespace Tfe.NetClient
+{
+    public class JsonKebabCaseNamingPolicy : JsonNamingPolicy
```
Comment (cmendible) [inline src/Tfe.NetClient/Features/JsonKebabCaseNamingPolicy.cs:6]: Nice!

**Review by cmendible — CHANGES_REQUESTED** [review]
Can you split the PR so we add the JsonKebabCaseNamingPolicy feature in one PR and the ConfigurationVersion and tests in other?

**Comment by mortengjesing** [comment]
Would like to avoid to implement the JsonPropertyName on ConfigurationVersions if possible. Why would you like to split it? Maybe I can come up with a solution to your concerns.

**Comment by cmendible** [comment]
Forget about splitting the PW I'm approving it as is.

**Pull request merged**